core/domain/play_later_services.py

```diff
@@ -0,0 +1,239 @@
+    from core.domain import learner_progress_services
+
+    play_later_activities_model = user_models.PlayLaterActivitiesModel.get(
+        user_id, strict=False)
+    if not play_later_activities_model:
+        play_later_activities_model = (
+            user_models.PlayLaterActivitiesModel(id=user_id))
+
+    subscribed_collection_ids = (
+        subscription_services.get_collection_ids_subscribed_to(user_id))
+    completed_collection_ids = (
+        learner_progress_services.get_all_completed_collection_ids(user_id))
+    incomplete_collection_ids = (
+        learner_progress_services.get_all_incomplete_collection_ids(user_id))
+
+    play_later_activities = get_play_later_activities_from_model(
+        play_later_activities_model)
+
+    if (collection_id not in subscribed_collection_ids and
+            collection_id not in completed_collection_ids and
+            collection_id not in incomplete_collection_ids):
+        if not position_to_be_inserted:
+            if len(play_later_activities.collection_ids) < 10:
+                play_later_activities.add_collection_id(collection_id)
+        else:
+            play_later_activities.remove_collection_id(collection_id)
+            play_later_activities.add_collection_id(
+                collection_id, position_to_be_inserted)
+        save_play_later_activities(play_later_activities)
+
+
+def remove_exploration_from_play_later_list(user_id, exploration_id):
+    """Removes the exploration from the play later list of the user
+    (if present).
+
+    Args:
+        user_id: str. The id of the user.
+        exploration_id: str. The id of the exploration to be removed.
+    """
+    play_later_activities_model = user_models.PlayLaterActivitiesModel.get(
+        user_id, strict=False)
+
+    if play_later_activities_model:
+        play_later_activities = get_play_later_activities_from_model(
+            play_later_activities_model)
+        if exploration_id in play_later_activities.exploration_ids:
+            play_later_activities.remove_exploration_id(exploration_id)
+            save_play_later_activities(play_later_activities)
+
+
+def remove_collection_from_play_later_list(user_id, collection_id):
+    """Removes the collection from the play later list of the user
+    (if present).
+
+    Args:
+        user_id: str. The id of the user.
+        collection_id: str. The id of the collection to be removed.
+    """
+    play_later_activities_model = user_models.PlayLaterActivitiesModel.get(
+        user_id, strict=False)
+
+    if play_later_activities_model:
+        play_later_activities = get_play_later_activities_from_model(
+            play_later_activities_model)
+        if collection_id in play_later_activities.collection_ids:
+            play_later_activities.remove_collection_id(collection_id)
+            save_play_later_activities(play_later_activities)
+
+
+def get_all_play_later_exploration_ids(user_id):
```

Comment:
Remember to think about speed -- if you're ever planning to get both exps and collections then you may just want one function that does a single RPC. OK to leave this as-is and refactor later if needed; just mentioning this in case.